### DISCUSSION TITLE
How to determine delay of a frame in a gif file?

### DISCUSSION THREAD

**Discussion by totalL2020** [ImageMagick/ImageMagick#2166] [Help] [answered]
I found information on how to set the delay, But how do you _get_ the delay of a frame?

**Comment by snibgo** [comment, marked as answer]
Create a test GIF:
```
f:\web\im>%IMG7%magick -delay 20 rose: -delay 70 rose: x.gif
```

Get the delays:
```
f:\web\im>%IMG7%magick x.gif -format %T\n info:
20
70
```

See https://www.imagemagick.org/script/escape.php

Or:
```
f:\web\im>%IMG7%magick x.gif -verbose info: |grep elay
  Delay: 20x100
  Delay: 70x100
```

Or: exiftool gives the total duration of the GIF, so you can calculate the average delay of each frame.

**Reply by totalL2020** [reply]
Thank you very much.
